coil-core/src/androidMain/kotlin/coil3/fetch/ContentUriFetcher.kt

```diff
@@ -30,30 +30,28 @@ internal class ContentUriFetcher(
     override suspend fun fetch(): FetchResult {
         val androidUri = data.toAndroidUri()
         val contentResolver = options.context.contentResolver
-        val inputStream = if (isContactPhotoUri(data)) {
+        val afd = if (isContactPhotoUri(data)) {
```

Comment:
Separate to this PR, but will `ImageDecoder.createSource(options.context.contentResolver, metadata.uri.toAndroidUri())` correctly handle this special case for a contacts image URI?

One of the issues with passing this data to `StaticImageDecoderDecoder` is we're coupling the source with the decoder.

Reply:
> Separate to this PR, but will `ImageDecoder.createSource(options.context.contentResolver, metadata.uri.toAndroidUri())` correctly handle this special case for a contacts image URI?
> 
> One of the issues with passing this data to `StaticImageDecoderDecoder` is we're coupling the source with the decoder.

Haven't test it, but now ImageDecoder use a afd that InputStream use for this case so it doesn't matter